ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -345,177 +144,58 @@ public void close() throws IOException {
       shutDownExecutorService(scheduler, config.updaterShutDownTimeoutMinutes, TimeUnit.MINUTES);
     }
     maybeUnsubscribeChangeTopic();
-    open.set(false);
+    cachedAccountService.close();
+    if (config.enableNewDatabaseForMigration) {
+      cachedAccountServiceNew.close();
+    }
   }
 
   @Override
   protected void checkOpen() {
-    if (!open.get()) {
-      throw new IllegalStateException("AccountService is closed.");
+    cachedAccountService.checkOpen();
+    if (config.enableNewDatabaseForMigration) {
+      cachedAccountServiceNew.checkOpen();
     }
   }
 
   ExecutorService getScheduler() {
     return scheduler;
   }
 
-  @Override
-  protected void publishChangeNotice() {
-    // TODO: can optimize this by sending the account/container payload as the topic message,
-    // so subscribers can update their cache and avoid the database query.
-    super.publishChangeNotice();
-  }
-
   @Override
   protected void onAccountChangeMessage(String topic, String message) {
-    if (!open.get()) {
-      // take no action instead of throwing an exception to silence noisy log messages when a message is received while
-      // closing the AccountService.
-      return;
-    }
-    try {
-      switch (message) {
-        case FULL_ACCOUNT_METADATA_CHANGE_MESSAGE:
-          logger.info("Processing message={} for topic={}", message, topic);
-          fetchAndUpdateCache();
-          logger.trace("Completed processing message={} for topic={}", message, topic);
-          break;
-        default:
-          accountServiceMetrics.unrecognizedMessageErrorCount.inc();
-          throw new RuntimeException("Could not understand message=" + message + " for topic=" + topic);
-      }
-    } catch (Exception e) {
-      logger.error("Exception occurred when processing message={} for topic={}.", message, topic, e);
-      accountServiceMetrics.remoteDataCorruptionErrorCount.inc();
-      accountServiceMetrics.fetchRemoteAccountErrorCount.inc();
-    }
+   cachedAccountService.onAccountChangeMessage(topic, message);
+   if (config.enableNewDatabaseForMigration) {
+     cachedAccountServiceNew.onAccountChangeMessage(topic, message);
+   }
```

Comment:
I think we can leave the `onAccountChangeMessage` (which is a `TopicListener` implementation)  empty in `MySqlAccountService` since each `CachedAccountService` also listen to the change.  Also, I think, since this listener is not registered (by calling `maybeSubscribeChangeTopic()` ) in MySqlAccountService, this might not be invoked.

Reply:
Sure, removed.